x-pack/plugins/apm/public/components/app/correlations/latency_correlations.tsx

```diff
@@ -199,23 +246,25 @@ function getSelectedDistribution(
 }
 
 function LatencyDistributionChart({
-  data,
+  overallData,
+  correlationsData,
   selectedSignificantTerm,
   status,
 }: {
-  data?: CorrelationsApiResponse;
+  overallData?: OverallLatencyApiResponse;
+  correlationsData?: CorrelationsApiResponse['significantTerms'];
   selectedSignificantTerm: SelectedSignificantTerm | null;
   status: FETCH_STATUS;
 }) {
   const theme = useTheme();
-  const xMax = Math.max(
-    ...(data?.overall?.distribution.map((p) => p.x ?? 0) ?? [])
-  );
+  const xMax = overallData?.overallDistribution
+    ? Math.max(...(overallData?.overallDistribution.map((p) => p.x ?? 0) ?? []))
+    : 0;
```

Comment:
Can't you make `getDistributionYMax` a bit more generic and reuse it here too? It looks like it does exactly the same thing.

```suggestion
  const xMax = getDistributionYMax(overallData?.overallDistribution)
```